ocaml/xapi/xapi_clustering.ml

```diff
@@ -562,9 +562,18 @@ module Watcher = struct
         | None ->
             ()
       )
-      | Error (InternalError "UPDATES.Timeout") ->
-          (* UPDATES.get timed out, this is normal, now retry *)
-          ()
+      | Error (InternalError "UPDATES.Timeout") -> (
+        (* UPDATES.get timed out, this is normal.  *)
+        match find_cluster_host ~__context ~host with
+        | Some ch ->
+            let cluster = Db.Cluster_host.get_cluster ~__context ~self:ch in
+            (* CA-395789: We send a query to xapi-clusterd to fetch the latest state
+               anyway in case there is a race and the previous update did not give the
+               most up-to-date information *)
+            on_corosync_update ~__context ~cluster ["routine updates"]
+        | None ->
+            ()
```

Comment:
Seems the logic here is the same as above case, how about add function for this? Then we can add comments before calling the function.